data-raw/R2R_TMH_habitat_inputs/cache_tmh_data.R

```diff
@@ -0,0 +1,290 @@
+library(tidyverse)
+
+source('data-raw/R2R_TMH_habitat_inputs/tmh_helper_functions.R')
+
+
+# file to cache Theoretical Max Habitat data objects 
+# and some exploratory plots to compare SIT existing 
+# to TMH. Currently just for Fall Run. 
+# TODO: adapt to add Spring and Winter Runs
+
+all_existing_and_tmh_data <- readRDS('data-raw/R2R_TMH_habitat_inputs/all_habitat_data_for_tmh_inputs_fall_run.rdata')
+
+all_hab_data_long <- all_existing_and_tmh_data |>
+  rename(spwn_acres_max = max_spawning_acres,
+         rear_acres_max = max_rearing_acres, 
+         flood_acres_max = max_floodplain_acres) |> 
+  pivot_longer(cols = c(spwn_acres_max:flood_acres_max), names_to = "metric") |>
+  separate(metric, c('hab', 'unit', 'lifestage'), "_") |>
+  mutate(max_hab = ifelse(lifestage == "max", "max_hab", NA),
+         lifestage = ifelse(lifestage == "max", NA, lifestage),
+         unit = ifelse(unit == "flow", "cfs", unit)) |> 
+  glimpse()
+
+
+# update DSMhabitat values ------------------------------------------------
+
+watersheds <- unique(all_hab_data_long$watershed)
+watersheds <- watersheds[!(watersheds %in%  c('North Delta', "South Delta"))]
+
+## spawning ----------------------------------------------------------------
+
+r_to_r_tmh_fr_spawn <- DSMhabitat::fr_spawn$biop_itp_2018_2019
+
+for(i in 1:length(watersheds)) {
+  ws = watersheds[i]
+  habitat = "spwn"
+  
+  max_hab_acres <- all_hab_data_long |> 
+    filter(watershed == ws & hab == habitat) |> 
+    filter(max_hab == "max_hab") |> 
+    pull(value)
+  
+  existing_acres <- existing_acres_fun(ws, 'spawn') 
+  
+  # Note:when existing SIT habitat is greater than TMH, we are still using TMH so the WUA would get scaled down
```

Comment:
The `existing_acres == 0` is more an issue with spawning but it seemed easier to add that error handling in all loops (especially for when I functionalize later). 

It's possible that the adjustment factor actually gets scaled down if existing acres is greater than max habitat. This is something that was discussed with Rene and we decided to use Max Hab which would mean the WUA would actually be smaller for Max Hab. This is a small subset but possible.

Reply:
Do you think we could add a conditional here? Like if existing_acers > TMH use TMH or something like that to make it explicit 

Also why would existing acres be 0 or NA?

Reply:
Existing acres could be NA or 0 if there is no spawning, example: 
`set_spawning_habitat("Upper-mid Sacramento River", "fr", 7500, month = 10)` returns `NA`

I don't think it makes sense to have a  conditional because the logic for the adjustment factor is the same.